__tests__/seed.js

```diff
@@ -73,81 +76,89 @@ async function seed(){
       name: 'test',
     });
 
-  //entity
-  {
-    await knex('entity')
-      .where('id', entity.id)
-      .del();
-    await knex('entity')
-      .insert({
-        id: entity.id,
-        type: entity.type,
-        name: entity.name,
-        wallet: entity.wallet,
-        password: entity.passwordHash,
-        salt: entity.salt,
-      });
-  }
+  // wallet
+  await knex('wallets.wallet')
+    .insert({
+      id: wallet.id,
+      type: wallet.type,
+      name: wallet.name,
+      password: wallet.passwordHash,
+      salt: wallet.salt,
+    });
+
+
+  // //entity
+  // {
+  //   await knex('entity')
+  //     .where('id', entity.id)
+  //     .del();
+  //   await knex('entity')
+  //     .insert({
+  //       id: entity.id,
+  //       type: entity.type,
+  //       name: entity.name,
+  //       wallet: entity.wallet,
+  //       password: entity.passwordHash,
+  //       salt: entity.salt,
+  //     });
+  // }
 
   //entity role
-  {
-    log.debug('clear role');
-    await knex('entity_role')
-      .insert([{
-        entity_id: entity.id,
-        role_name: 'list_trees',
-        enabled: true,
-      },{
-        entity_id: entity.id,
-        role_name: 'manage_accounts',
-        enabled: true,
-      },{
-        entity_id: entity.id,
-        role_name: 'accounts',
-        enabled: true,
-      }]);
-  }
+  log.debug('clear role');
+  await knex('entity_role')
```

Comment:
The above deletion (entity) leads to error here, on my side, a foreign key exception, I thought you should throw error too because if you delete the entity insert statement above, them here should not be able to pass, the entity_id here violate the foreign key rule..?

Reply:
Ah you're right, we have to start from a cleared, empty database. For now I will put back the entity table insertion and deletion just so we can continue to use and test the entity_role table. In the future, we will not be manipulating any of the tables in the public schema (including any entity_ related tables) since we will be accessing it through an API but this is just a placeholder for now until we build out that infrastructure. Maybe as a way to "mock" the API call, we can consider just testing with a prepopulated object that includes all the data we will need from an entity_role table.